MatchServer/GunGame.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include <MXml.h>
+#include <map>
+#include <vector>
+#include "MZFileSystem.h"
+struct GGSet
+{
+	UINT  WeaponSet[3];
+};
+
+class GunGame
+{
+private:
+	bool ParseXML_ItemSet(MXmlElement& elem, GGSet* Node);
+	std::map<int,std::vector<GGSet*>> m_GGSet;
```

Comment:
Having a container of raw pointers and no copy constructor is a terrible MAIET-ism.

Some reasonable things to do instead are:

* Having a container of normal value types, i.e. `std::vector<GGSet>` in this case

* Having a container of `std::unique_ptr<T>` if you really need them to be pointers (which you don't, in this case)

Also, I'm not sure why you have a map, or why you store indices in the XML. It seems like the only thing you use this thing for is retrieving a random list of `GGSet`s, so `std::vector<std::vector<GGSet>>` seems like a more obvious choice.